plugins/content_distribution/providers/facebook/lib/FacebookDistributionEngine.php

```diff
@@ -113,22 +123,52 @@ protected function doUpdate(KalturaDistributionUpdateJobData $data)
 			throw new Exception("Failed to update facebook video , reason:".$e->getMessage());
 		}
 
-		// first delete the captions that were already distributed
-		while ($mediaFile = array_pop($data->mediaFiles))
+		$tags = FacebookDistributionEngine::helperGetTags($data->providerData->fieldValues);
+		$this->handleTags($data->distributionProfile->pageAccessToken, $tags ,$data->entryDistribution->remoteId);
+		$this->handleCaptions($data);
+
+		return true;
+	}
+
+	private function handleTags($pageAccessToken, $tags, $videoId)
+	{
+		if(empty($tags))
+			return;
+
+		try
+		{
+			FacebookGraphSdkUtils::updateTags($this->appId, $this->appSecret, $pageAccessToken, $tags, $videoId);
+		}
+		catch (Exception $e)
 		{
-			$this->deleteCaption($data->distributionProfile, $mediaFile->remoteId, $data->entryDistribution->remoteId);
+			throw new Exception("Failed to update facebook tags , reason:".$e->getMessage());
 		}
-		// last add all the captions available
-		if(isset($data->providerData->captionsInfo))
+	}
+
+	private function handleCaptions(KalturaDistributionUpdateJobData $data)
+	{
+		try
 		{
-			foreach ($data->providerData->captionsInfo as $captionInfo)
+			// first delete the captions that were already distributed
+			while ($mediaFile = array_pop($data->mediaFiles))
```

Comment:
Why not use the common way of foreach on mediaFiles

Reply:
Not my code I just moved it around but I think he used that way in case mediaFiles is null so this way he dont need to check it before calling the foreach